Tools/build/check_warnings.py

```diff
@@ -191,7 +194,10 @@ def get_unexpected_improvements(
     """
     unexpected_improvements = []
     for rule in ignore_rules:
-        if not rule.ignore_all and rule.file_path not in files_with_warnings.keys():
+        if (
+            not rule.ignore_all
+            and rule.file_path not in files_with_warnings.keys()
```

Comment:
This function is typed incorrectly, `files_with_warnings: set[IgnoreRule],` needs to be `dict[...]` or it should take `files_with_warnings.keys()` for a set-like interface (in which case you would need to use `Set[...]` from `collections.abc`.
```suggestion
            and rule.file_path not in files_with_warnings
```